### ISSUE TITLE
nm-util-dev 1.8.4-2 dropped gir1.2-nm-1.0 on debian

### ISSUE THREAD

**Issue by legrostdg** [firecat53/networkmanager-dmenu#47]
On Debian sid, since nm-util-dev 1.8.4-2, networkmanager-dmenu stopped working. The reason is it switched from gir1.2-nm-1.0 to gir1.2-networkmanager-1.0. Should we convert "NM" to "NetworkManager"? Or just (temporary) add gir1.2-nm-1.0 to the requirements?

**Comment by firecat53** [comment]
I think this is an issue for the Debian package maintainer. I can't help you with this! This repository is just the networkmanager_dmenu python script.

**Comment by legrostdg** [comment]
Currently there's no Debian package for networkmanager_dmenu. The README lists nm-util-dev as a dependency, but not gir1.2-nm-1.0 (which used to be a dependency of nm-util-dev). The README should at least list gir1.2-nm-1.0 as a requirement for building networkmanager_dmenu on Debian.

**Comment by legrostdg** [comment]
But if upstream nm switched to gir1.2-networkmanager-1.0, I guess the right thing to do would be to convert networkmanager_dmenu to use `NetworkManager` instead of `NM` in its code.

**Comment by legrostdg** [comment]
But if upstream nm switched to gir1.2-networkmanager-1.0, I guess the right thing to do would be to convert networkmanager_dmenu to use `NetworkManager` instead of `NM` in its code.

**Comment by firecat53** [comment]
I misunderstood you. I thought you meant the package file name changed, but you're saying that the actual package import name changed. I'll have to look into this more.

**Comment by firecat53** [comment]
I added gir1.2-nm-1.0 to the README as an added dependency for Debian Sid. I poked at the code awhile to see if I could remove the `NM` dependency and just use the `NMClient` and `NetworkManager` objects. There didn't seem to be an easy conversion. The missing piece that I couldn't easily find was the equivalent of `Client.get_connections`. There may have been more issues, but I stopped after I couldn't easily solve that one. Closed by eefd2c34. Thanks for the report.
